### ISSUE TITLE
Cpu Usage percent when run video inference

### ISSUE THREAD

**Issue by NgTuong** [vita-epfl/openpifpaf#368]
Hi @svenkreiss , I run video inference on RTX2080Ti 20 cores, I used batch size = 1, long_egde=512. Gpu memory ~ only 1.2GB but cpu usage ~ 1000%cpu. I wonder that what is the cause of problem?

**Comment by svenkreiss** [comment]
Seems right to me. The decoder runs on the CPU. GPU memory is only high during training. During prediction, we deactivate the gradient graph which is unnecessary and saves a lot fo resources (see `torch.no_grad`).

**Comment by NgTuong** [comment]
Thanks @svenkreiss ,  What do you have idea to optimize when the decoder runs on the CPU?
